src/index.js

```diff
@@ -1,21 +1,60 @@
+import checkTask from './module/task-survey.js';
+import descriptionOnFocus from './module/focus-functionality.js';
+// eslint-disable-next-line import/no-cycle
+import {
+  addNewItem, clearAll, updateInput, removeItem,
+} from './module/add-remove-implementation.js';
+
 import './style.css';
 
-const toDoItems = [
-  { description: 'Do the chores', completed: false, index: 0 },
-  { description: 'Take a nap', completed: false, index: 1 },
-  { description: 'Do Microverse Work', completed: false, index: 2 },
-];
+// SET TODO ARR
+let toDoArr = [];
+export const setToDoArr = () => {
+  if (JSON.parse(localStorage.getItem('myToDos'))) {
+    toDoArr = JSON.parse(localStorage.getItem('myToDos'));
+  } else {
+    toDoArr = [];
+  }
+};
+setToDoArr();
 const listWrap = document.querySelector('.todo-list-wrap');
 // IMPLEMENT TO DO ITEMS TO SCREEN
-const implementToDoItems = () => {
-  toDoItems.forEach((task) => {
-    listWrap.innerHTML += `<li class="list-item" data-key="${task.index}">
+export const implementToDoItems = (arr) => {
+  arr.forEach((item) => {
+    let isCompleted = '';
+    let completedClass = '';
+    if (item.completed) {
+      isCompleted = 'checked';
+      completedClass = 'completed-item';
+    }
+    listWrap.innerHTML += `<li class="list-item ${completedClass}" data-key="${item.index}" draggable="true">
       <div class="list-item-check-name">
-        <button type="button" class="check-item" data-key="${task.index}"></button>
-        <input class="description-input" type="text" value="${task.description}">
+         <input type="checkbox" class="check-item" data-key="${item.index}" ${isCompleted}>
+  <input class="description-input" data-key="${item.index}" type="text" value="${item.description}">
       </div>
-      <button data-key="${task.index}" class="move-item"><i class="fas fa-ellipsis-v"></i></button>
+      <button type="button" data-key="${item.index}" class="move-item"><i class="fas fa-ellipsis-v"></i></button>
+      <button type="button" data-key="${item.index}" class="remove-item"><i class="far fa-trash-alt"></i></button>
     </li>`;
   });
 };
-implementToDoItems();
+implementToDoItems(toDoArr);
+// CHECK ITEMS
+const toDoChecksInputs = [...document.querySelectorAll('.list-item .check-item')];
+toDoChecksInputs.forEach((item) => item.addEventListener('change', () => checkTask(item)));
+// FOCUS ITEMS
+const toDoDescriptionsInputs = [...document.querySelectorAll('.list-item .description-input')];
+toDoDescriptionsInputs.forEach((item) => item.addEventListener('focusin', (event) => descriptionOnFocus(event, item)));
+toDoDescriptionsInputs.forEach((item) => item.addEventListener('focusout', (event) => descriptionOnFocus(event, item)));
+const toDoItems = [...document.querySelectorAll('.list-item')];
+
+// git ADD ITEM
+const addNewButton = document.querySelector('.add-new-btn');
+addNewButton.addEventListener('click', (event) => addNewItem(event, toDoArr, toDoItems, listWrap));
```

Comment:
- [ ] Kindly when I tried to add a task, it adds two tasks one with the data and another empty one, this issue may be because the event is run twice so when you click enter or the button it calls the `addNewItem` method twice. 

![image](https://user-images.githubusercontent.com/35535130/161123489-f289d92a-93ed-45ea-9361-1030051a38ba.png)

Reply:
Thanks for the review changes made